integration/walletextension/wallet_extension_test.go

```diff
@@ -427,6 +429,25 @@ func TestCannotSubscribeOverHTTP(t *testing.T) {
 	}
 }
 
+func TestSuccessGasEstimate(t *testing.T) {
+	createWalletExtension(t)
+	accountAddr, _ := registerPrivateKey(t)
+	callMsg := datagenerator.CreateCallMsg()
+	callMsg.From = accountAddr
+
+	callMsgBytes, err := json.Marshal(callMsg)
+	if err != nil {
+		t.Fatalf("unable to marshal CallMsg - %s", err)
+	}
+
+	callMsgHex := hexutil.Encode(callMsgBytes)
+	getBalanceJSON := makeHTTPEthJSONReqAsJSON(rpc.RPCEstimateGas, []string{callMsgHex, latestBlock})
+
+	if getBalanceJSON[walletextension.RespJSONKeyResult].(string) != "0x12a05f200" {
+		t.Fatalf("unexpected gas")
```

Comment:
@otherview Can you print the expected vs actual gas here? e.g. it will be interesting to see if the actual gas is `0x0`.

Reply:
If you use the `testify` library that is used in some places in our code (perhaps only by me, thought it was standard...) you get that output for free. I.e. `assert.Equals(result, expected, "unexpected gas")`

Reply:
Yep, in the plans when the gas estimation is dynamic. I don't expect it to change atm.

Reply:
I think you guys are in the right direction. Changed the errors to be more narrowed (not a catch all) and used the `testify` package too!